### PR TITLE
Simple JAX-RS resource for CRUD operations on cluster configuration

### PR DESCRIPTION
**Pull request by joschi**
This PR adds a relatively simple JAX-RS resource for performing CRUD operations on the Graylog cluster configuration.

### CODE REVIEW

File: graylog2-server/src/main/java/org/graylog2/rest/resources/system/ClusterConfigResource.java [hunk 1]
```diff
@@ -0,0 +1,152 @@
...
+import javax.ws.rs.core.MediaType;
+import javax.ws.rs.core.Response;
+import java.io.IOException;
+import java.io.InputStream;
+import java.util.Set;
+
+import static java.util.Objects.requireNonNull;
+
+@Api(value = "System/ClusterConfig", description = "Graylog Cluster Configuration")
+@RequiresAuthentication
```
Comment (bernd) [inline graylog2-server/src/main/java/org/graylog2/rest/resources/system/ClusterConfigResource.java]: Please also use the existing `RestPermissions.CLUSTER_CONFIG_*` permissions for the different endpoints.

**Comment by bernd** [comment]
To avoid excessive HTTP calls from the web interface, we need one endpoint that takes a list of cluster config classes and returns the values for them. If a given class does exist but there is no entry for it in the database, it should return the default configuration for that entry.

We might have to add an interface for this with a `default()` method to ensure that cluster config objects have a default config.

**Comment by joschi** [comment]
@bernd I'd like to tackle those issues/improvements in a follow-up PR to keep things quickly reviewable.

Regarding the default configuration, what would be the returned value if there is no (sane) default?

**Comment by bernd** [comment]
> I'd like to tackle those issues/improvements in a follow-up PR to keep things quickly reviewable.

:+1: 

> Regarding the default configuration, what would be the returned value if there is no (sane) default?

I think every object in the cluster config that is a "system config" object like retention/rotation or email config has some default values. Otherwise it is not really usable.

After looking at the existing configs, I think there is none that has no default config. Even the `ClusterId` object, its default can be a new UUID every time the default is requested.

**Comment by joschi** [comment]
> Even the `ClusterId` object, its default can be a new UUID every time the default is requested.

But exactly that use case is a problem. If `ClusterId` returns a random UUID by default which is not persisted anywhere, it'll lead to funny effects because different parts of the application will use different Cluster IDs.

**Comment by bernd** [comment]
> But exactly that use case is a problem. If ClusterId returns a random UUID by default which is not persisted anywhere, it'll lead to funny effects because different parts of the application will use different Cluster IDs.

Okay, I see. But is it possible that this is not set in the database? This should be written to the DB as early as possible, no?

If this is a problem, we might have to extend the interface to allow different types of configs. One with default values, one without.

File: graylog2-server/src/main/java/org/graylog2/rest/resources/system/ClusterConfigResource.java (hunk 1 above)
Comment (joschi) [inline graylog2-server/src/main/java/org/graylog2/rest/resources/system/ClusterConfigResource.java]: :white_check_mark:

**Comment by joschi** [comment]
@bernd The possibility to set a default configuration object will be implemented in the scope of another PR.

**Comment by bernd** [comment]
LGTM :+1:

**Pull request merged**